app/services/bulk_hosts_manager.rb

```diff
@@ -19,6 +19,13 @@ def build(reboot: false)
     end
   end
 
+  def reassign_hostgroups(hostgroup)
+    @hosts.each do |host|
+      host.hostgroup = hostgroup
+      host.save(:validate => false)
+    end
```

Comment:
This triggers a ton of saves and will be slow. A little research shows you can do:
```suggestion
    @hosts.update_all(hostgroup: hostgroup)
```

This doesn't update the `updated_at`, but `hostgroup` doesn't have such a column. It also doesn't trigger callbacks, so it may break (web)hooks. Have you considered this?

Reply:
I didn't try to change any existing back-end code other than moving it around to support the API call.

Reply:
`save(:validate => false)` is also the way the legacy page does it.. I don't see any harm in doing it the same way for now.

Reply:
I think that's fair, but it's also worth keeping an eye on improving performance.